### ISSUE TITLE
NullPointerException in io.quarkus.jackson.runtime.ObjectMapperProducer_ProducerMethod_objectMapper

### ISSUE THREAD

**Issue by zailushangde** [quarkusio/quarkus#24100]
### Describe the bug

We have a function to transform a string to JsonNode, which is done using the default object mapper, without any customization. 

Here is what the class signature looks like

```
@ApplicationScoped
class LegacyValueBackfiller(private val objectMapper: ObjectMapper) {}
```

However, **sometimes** it throws an NPE, and here are the frames

```
exception: {
    refId:         1,
    exceptionType: 'java.lang.NullPointerException',
    message:       null,
    frames: [
      {
        class:  'io.quarkus.jackson.runtime.ObjectMapperProducer_ProducerMethod_objectMapper_7d4fe425e6f58abcbdd78addaab8a43fdb7ab3ae_Bean',
        method: 'get',
      },
      {
        class:  'io.quarkus.jackson.runtime.ObjectMapperProducer_ProducerMethod_objectMapper_7d4fe425e6f58abcbdd78addaab8a43fdb7ab3ae_Bean',
        method: 'get',
      },
      {
        class:  '{MY_PATH}.LegacyValueBackfiller_Bean',
        method: 'create',
      },
      {
        class:  '{MY_PATH}.LegacyValueBackfiller_Bean',
        method: 'create',
      },
      {
        class:  'io.quarkus.arc.impl.AbstractSharedContext',
        method: 'createInstanceHandle',
        line:   101,
      },
      {
        class:  'io.quarkus.arc.impl.AbstractSharedContext$1',
        method: 'get',
        line:   29,
      },
      {
        class:  'io.quarkus.arc.impl.AbstractSharedContext$1',
        method: 'get',
        line:   26,
      },
      {
        class:  'io.quarkus.arc.impl.LazyValue',
        method: 'get',
        line:   26,
      },
      {
        class:  'io.quarkus.arc.impl.ComputingCache',
        method: 'computeIfAbsent',
        line:   69,
      },
      {
        class:  'io.quarkus.arc.impl.AbstractSharedContext',
        method: 'get',
        line:   26,
      },
      {
        class:  'io.quarkus.arc.impl.ClientProxies',
        method: 'getApplicationScopedDelegate',
        line:   18,
      },
      {
        class:  '{MY_PATH}.LegacyValueBackfiller_ClientProxy',
        method: 'arc$delegate',
      },
     {
        class:  '{MY_PATH}.LegacyValueBackfiller_ClientProxy',
        method: '{MY_FUNCTION}',
      },
      ...
```

### Actual behavior

Sometimes a null pointer exception is thrown, which does not occur at the start of the application, etc ...

### Output of `uname -a` or `ver`

Linux {MY_POD} 5.4.0-1063-aws #66-Ubuntu SMP Wed Jan 12 17:49:45 UTC 2022 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "11.0.14.1" 2022-02-08 OpenJDK Runtime Environment 18.9 (build 11.0.14.1+1) OpenJDK 64-Bit Server VM 18.9 (build 11.0.14.1+1, mixed mode, sharing)

### Quarkus version or git rev

2.7.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 7.2

**Comment by geoand** [comment]
We'll need a sample application that exhibits the problematic behavior in order to investigate.

**Comment by zailushangde** [comment]
> We'll need a sample application that exhibits the problematic behavior in order to investigate.

it is hard to reproduce it though

**Comment by geoand** [comment]
> it is hard to reproduce it though

Yeah I understand.

However just looking at the logs, it's impossible to even guess about what might be the root cause of the problem.

**Comment by cjbooms** [comment]
As we can't reproduce this bug, we have worked around it by catching and retrying any requests that trigger it.

It feels to me like some sort of intermittent non-atomic context refresh is causing the issue. But that's purely based on looking at logs.

I've suggested the team place some additional logging in all exception mappers to try and better track this issue in our production code. We will revert back to this issue with any findings we have. For now, we don't have enough to go on to attempt to create a standalone app to repo it.
